### DISCUSSION TITLE
INP001 triggered in constants directory

### DISCUSSION THREAD

**Discussion by remchuk** [astral-sh/ruff#15134] [Q&A] [answered]
Hey there fellows,
I apologize in advance if this post doesn't belong in here, I'm quite new with using Ruff.

I have quite a large python project with flask, with a main app (a directory called A with many directories inside).

Inside of A there are 5 directory:

- constants
- services
- utils
- dal
- resources

Due to the fact it is a Flask project, we have `__init__.py` under A.
Is `__init__.py` in all of the directories under A is necessary? is it an overkill? 

(we run `flask --app ./A/ run`)

Is this intended? If feel like it is a false positive.

example:
```
A/
|
---- constants/
     |  
     ------- sites.py
---- services/
----- utils/
----- dal/
----- resources/

```

In sites.py I have the following:

```
NY = "new-york"
TX = "texas"
AZ = "arizona"
```

but  `NY = "new-york"` prompts `INP001`.

You are much wiser and more experiences people of python, any advice? is it false positive?
I found some similer issues but all of them doesn't seem to solve this issue.

**Comment by MichaReiser** [comment, marked as answer]
Hi @remchuk 

Thanks for the detailed write-up. The Python documentation has a section that explains [when namespace packages](https://packaging.python.org/en/latest/guides/packaging-namespace-packages/) are the most useful. You'll know best if this applies to your project or if you have other reasons why the "caveats of namespace packages" are okay in your project. If you're uncertain, I'd recommend you to stick to regular python packages (add `__init__.py` files).
